src/storage/createRestStorage.ts

```diff
@@ -0,0 +1,283 @@
+
+		}).then(function() {
+			return {
+				successfulData: items,
+				type: StoreOperation.Put
+			};
+		});
+	},
+
+	add(this: Storage<{}, {}>, items: {}[], options?: CrudOptions): Promise<UpdateResults<{}>> {
+		return this.put(items, options).then(function(result) {
+			result.type = StoreOperation.Add;
+			return result;
+		});
+	},
+
+	delete(this: Storage<{}, {}>, ids: string[]): Promise<UpdateResults<{}>> {
+		if (!ids || ids.length === 0) {
+			return Promise.reject(new Error('ids can\'t be empty.'));
+		}
+		const state = instanceStateMap.get(this);
+		let uri = state.target;
+
+		if (ids.length === 1) {
+			uri += '/' + ids[0];
+		}
+		else {
+			uri += '?' + state.idSerializer(ids);
+		}
+
+		return <Promise<any>> request.delete(uri).then(function() {
+			return {
+				successfulData: ids,
+				type: StoreOperation.Delete
+			};
+		});
+	},
+
+	patch(this: Storage<{}, {}>, updates: { id: string; patch: Patch<{}, {}> }[]): Promise<UpdateResults<{}>> {
+		if (!updates || updates.length === 0) {
+			return Promise.reject(new Error('updates can\'t be empty.'));
+		}
+		const state = instanceStateMap.get(this);
+
+		const patchRequest = (id: string, data: string) => {
+			return request(state.target + '/' + id, {
+				data: data,
+				method: 'PATCH'
+			}).then(() => {
+				return this.get([id]);
+			}).then(function(items) {
+				return {
+					successfulData: items,
+					type: StoreOperation.Patch
+				};
+			});
+		};
+
+		const getNPutRequest = () => {
+			return this.get(updates.map(({id}) => id )).then((items) => {
+				const updatedItems = items.map((item) => {
+					const _id = this.identify(item)[0];
+					const patch = find(updates, ({id}) => id === _id).patch;
+					return patch.apply(item);
+				});
+				return this.put(updatedItems);
+			}).then(function(result) {
+				result.type = StoreOperation.Patch;
+				return result;
+			});
+
+		};
+		if (updates.length === 1) {
+			// user option takes precedence
+			const patchSerializer = getPatchSerializer(state.acceptPatch);
+			if (patchSerializer) {
+				return <Promise<any>> patchRequest(updates[0].id, patchSerializer(updates[0].patch));
+			}
+
+			return <Promise<any>> state.initPromise.then(function(acceptPatch: string) {
+				return getPatchSerializer(acceptPatch);
+			}, function() {
+				return null;
+			}).then(function(patchSerializer) {
+				if (patchSerializer) {
+					return patchRequest(updates[0].id, patchSerializer(updates[0].patch));
+				}
+				else {
+					return <Promise<any>> getNPutRequest();
+				}
+			});
+		}
+		else {
+			return getNPutRequest();
+		}
+	},
+
+	isUpdate(this: Storage<{}, {}>, item: {}): Promise<{ isUpdate: boolean; item: {}; id: string }> {
+		const id = this.identify(item)[0];
+		return this.get([id]).then(function(items) {
+			return {
+				id: id,
+				item: item,
+				isUpdate: items.length > 0
+			};
+		});
+	}
+}, <T, O>(instance: Storage<T, O>, {target,
+									createIdTarget,
+									idSerializer = defaultIdSerializer,
```

Comment:
I agree. I am making changes for this. For `get` and `delete`, I will repurpose `idSerializer` as the indicator, and for `put`, I will add another option, since `requestSerializer` is being used in multiple scenarios.